tests/chainer_tests/function_hooks_tests/test_timer.py

```diff
@@ -99,4 +101,32 @@ def test_backward_gpu(self):
         self.check_backward(cuda.to_gpu(self.x), cuda.to_gpu(self.gy))
 
 
+class TestTimerSummary(unittest.TestCase):
+
+    def setUp(self):
+        self.h = function_hooks.TimerHook()
+        self.f = functions.Exp()
+        self.f.add_hook(self.h)
+        self.x = numpy.random.uniform(-0.1, 0.1, (3, 5)).astype(numpy.float32)
+
+    def test_summary(self):
+        x = self.x
+        self.f(chainer.Variable(x))
+        self.f(chainer.Variable(x))
+        self.assertEqual(2, len(self.h.call_history))
+        self.assertEqual(1, len(self.h.summary()))
+
+    def test_print_summary(self):
+        x = self.x
+        self.f(chainer.Variable(x))
+        self.f(chainer.Variable(x))
+        io = six.StringIO()
+        self.h.print_summary(file=io)
+        expect = '''\AFunctionName  ElapsedTime  Occurrence
+ +Exp +[0-9.\-e]+.s +[0-9]+$
+'''
+        actual = io.getvalue()
+        self.assertTrue(re.match(expect, actual), actual)
```

Comment:
How about using `self.assertRegexpMatches`?
Also, prefix the regexp literal with `r`.